include/gridtools/fn/sid_neighbor_table.hpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2022, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include <cstddef>
+#include <type_traits>
+
+#include "../common/array.hpp"
+#include "../fn/unstructured.hpp"
+#include "../sid/concept.hpp"
+
+namespace gridtools::fn::sid_neighbor_table {
+    namespace sid_neighbor_table_impl_ {
+        template <class IndexDimension, class NeighborDimension, std::size_t MaxNumNeighbors, class Sid>
+        struct sid_neighbor_table {
+            Sid sid;
```

Comment:
The most important comment got lost because I refreshed: this does not work on GPUs. You have to store the trivially copyable ptr holder and strides here, not the SID directly. That is, this struct should contain the return values of `sid::get_origin` and `sid::get_strides`.

Reply:
Changed it. What is the easiest way to test at least the compilation on the GPU?

Reply:
Here is an example of the pattern that we use for testing functions on device https://github.com/GridTools/gridtools/blob/master/tests/unit_tests/common/test_int_vector.cu

Reply:
I added a new file for testing on the device, could you guys take a look at it?